runtime/permissions.rs

```diff
@@ -240,11 +242,38 @@ impl AsRef<str> for EnvDescriptor {
   }
 }
 
-#[derive(Clone, Eq, PartialEq, Hash, Debug, Default)]
-pub struct RunDescriptor(pub String);
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub enum RunDescriptor {
+  Name(String),
+  Path(PathBuf),
+}
+
+impl FromStr for RunDescriptor {
```

Comment:
Great changes in this file, having explicit `Default` implementations is also more readable in this case.